src/psyclone/dynamo0p3.py

```diff
@@ -3229,13 +3578,17 @@ def fs_compulsory_field(self, function_space):
     def basis(self, function_space):
         '''add basis function information for this function space to the
         argument list'''
-        basis_name = get_fs_basis_name(function_space)
+        basis_name = get_fs_basis_name(function_space,
+                                       qr_var=self._kern.qr_name,
+                                       on_space=self._kern.eval_fspace)
         self._arglist.append(basis_name)
 
     def diff_basis(self, function_space):
         '''add differential basis information for the function space to the
         argument list'''
-        diff_basis_name = get_fs_diff_basis_name(function_space)
+        diff_basis_name = get_fs_diff_basis_name(
```

Comment:
document args in parent diff_basis method.

Reply:
Done.